org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/AllBreakpointsEnablementAction.java

```diff
@@ -0,0 +1,126 @@
+ *******************************************************************************/
+package org.eclipse.debug.internal.ui.actions.breakpoints;
+
+import org.eclipse.core.resources.IMarkerDelta;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.IBreakpointManager;
+import org.eclipse.debug.core.IBreakpointsListener;
+import org.eclipse.debug.core.model.IBreakpoint;
+import org.eclipse.debug.internal.ui.DebugUIPlugin;
+import org.eclipse.debug.internal.ui.actions.AbstractRemoveAllActionDelegate;
+import org.eclipse.debug.internal.ui.actions.ActionMessages;
+import org.eclipse.jface.action.IAction;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.dialogs.MessageDialogWithToggle;
+import org.eclipse.jface.preference.IPreferenceStore;
+import org.eclipse.ui.IWorkbenchWindow;
+
+abstract class AllBreakpointsEnablementAction extends AbstractRemoveAllActionDelegate implements IBreakpointsListener {
+
+	private final boolean fEnable;
+	private final String fPreferenceName;
+	private final String fLabel1;
+	private final String fLabel2;
+
+	AllBreakpointsEnablementAction(boolean enable, String preferenceName, String label1, String label2) {
+		this.fEnable = enable;
+		this.fPreferenceName = preferenceName;
+		this.fLabel1 = label1;
+		this.fLabel2 = label2;
+	}
+
+	@Override
+	protected boolean isEnabled() {
+		return DebugPlugin.getDefault().getBreakpointManager().getBreakpoints().length > 0;
+	}
+
+	@Override
+	public void breakpointsAdded(IBreakpoint[] breakpoints) {
+		update();
+	}
+
+	@Override
+	public void breakpointsChanged(IBreakpoint[] breakpoints, IMarkerDelta[] deltas) {
+		update();
+	}
+
+	@Override
+	public void breakpointsRemoved(IBreakpoint[] breakpoints, IMarkerDelta[] deltas) {
+		if (getAction() != null) {
+			update();
+		}
+	}
+
+	@Override
+	protected void initialize() {
+		DebugPlugin.getDefault().getBreakpointManager().addBreakpointListener(this);
+	}
+
+	@Override
+	public void dispose() {
+		DebugPlugin.getDefault().getBreakpointManager().removeBreakpointListener(this);
+		super.dispose();
+	}
+
+
+	@Override
+	public void run(IAction action) {
+		final IBreakpointManager breakpointManager = DebugPlugin.getDefault().getBreakpointManager();
+		final IBreakpoint[] breakpoints = breakpointManager.getBreakpoints();
+		if (breakpoints.length < 1) {
+			return;
+		}
+		IWorkbenchWindow window = DebugUIPlugin.getActiveWorkbenchWindow();
+		if (window == null) {
+			return;
+		}
+		IPreferenceStore store = DebugUIPlugin.getDefault().getPreferenceStore();
+		boolean prompt = store.getBoolean(this.fPreferenceName);
+		boolean proceed = true;
+		if (prompt) {
+			MessageDialogWithToggle mdwt = MessageDialogWithToggle.openYesNoQuestion(window.getShell(),
```

Comment:
I think I'll "merge the preference and dialog into one, enabling and disabling all breakpoints.